### ISSUE TITLE
Root not found in tree error hangs the node (Sepolia long run)

### ISSUE THREAD

**Issue by rodrigo-o** [lambdaclass/lambda_ethereum_consensus#1309]
This is something that could happen along the node executions but is easily reproducible on the initial setup. After the initial sync and some time running (a couple of minutes) an initial pruning kicks in and for some reason it generates a slot in the future error, here is an example log:

```
INFO 17:43:12.001 [StateDb] Pruning started. slot=5963552
INFO 17:43:12.001 [BlobDb] Pruning started. slot=5832480
INFO 17:43:12.001 [BlockDb] Pruning started. slot=5963552
INFO 17:43:12.001 [BlobDb] Pruning finished. 0 blobs removed. 
INFO 17:43:12.003 [Libp2p] Slot transition slot=5963616
INFO 17:43:12.009 [BlockDb] Pruning finished. 52 blocks removed. 
INFO 17:43:12.019 [StateDb] Pruning finished. 52 states removed. 
INFO 17:43:12.684 [Gossip] Block received, block.slot: 5963616. 
INFO 17:43:24.002 [Libp2p] Slot transition slot=5963617
INFO 17:43:24.216 [Gossip] Block received, block.slot: 5963617. 
INFO 17:43:29.977 [Fork choice] Adding new block slot=5963613 root=0x3de..fc05
INFO 17:43:30.204 [Fork choice] Block processed. Recomputing head. 
ERROR 17:43:30.204 GenServer LambdaEthereumConsensus.Libp2pPort terminating
** (RuntimeError) Parent 14C35B27D611342466C990ABE85147AFCDA13DBE2D5C05C4B3155309CF94FFFD not found in tree
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/simple_tree.ex:84: LambdaEthereumConsensus.ForkChoice.Simple.Tree.get_children!/2
    (lambda_ethereum_consensus 0.1.0) lib/types/store.ex:147: Types.Store.get_children/2
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/head.ex:90: LambdaEthereumConsensus.ForkChoice.Head.filter_block_tree/4
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/head.ex:85: LambdaEthereumConsensus.ForkChoice.Head.get_filtered_block_tree/1
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/head.ex:14: LambdaEthereumConsensus.ForkChoice.Head.get_head/1
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:269: LambdaEthereumConsensus.ForkChoice.recompute_head/1
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:62: anonymous fn/1 in LambdaEthereumConsensus.ForkChoice.on_block/2
    (telemetry 1.3.0) /home/admin/lambda_ethereum_consensus/deps/telemetry/src/telemetry.erl:324: :telemetry.span/3
Last message: {#Port<0.16>, {:data, <<50, 145, 166, 1, 10, 36, 55, 100, 49, 52, 49, 57, 53, 49, 45, 97, 100, 56, 50, 45, 52, 50, 101, 102, 45, 97, 102, 102, 101, 45, 48, 57, 102, 54, 50, 55, 101, 97, 57, 56, 102, 101, 16, 1, 26, 229, ...>>}} 
INFO 17:43:30.227 [Optimistic Sync] Waiting 10.0 seconds to discover some peers before requesting blocks. 
INFO 17:43:30.227 [Fork choice] Adding new block slot=5963613 root=0x3de..fc05
ERROR 17:43:30.280 [Fork choice] Failed to add block: block is from the future slot=5963613 root=0x3de..fc05
ERROR 17:43:30.280 [PendingBlocks] Saving block as invalid block is from the future slot=5963613 root=0x3de..fc05
INFO 17:43:36.001 [Libp2p] Slot transition slot=5963618
INFO 17:43:40.228 [Optimistic sync] Performing optimistic sync between slots 5963489 and 5963618, for a total of 130 slots. 
```

This happens to kick in sync again and and may trigger #1308 issue.

**Comment by rodrigo-o** [comment]
This was not related to pruning as I originally thought. Here is a particular example where pruning wasn't involved. Something to be mindful about is that the error was pointing to a block that we added at the start of the following log, and that was correctly processed and not reorged in Sepolia (https://sepolia.etherscan.io/block/6790864#consensusinfo)

```
INFO 21:08:40.621 [Fork choice] Adding new block slot=6000639 root=0xb3a..dbe9
INFO 21:08:40.775 [Fork choice] Block processed. Recomputing head. 
INFO 21:08:40.785 [Fork choice] Added new block slot=6000639 root=0xb3a..dbe9
INFO 21:08:40.786 [Fork choice] Adding new block slot=6000640 root=0xcae..a8f3
INFO 21:08:41.061 [Fork choice] Block processed. Recomputing head. 
INFO 21:08:41.067 [Fork choice] Added new block slot=6000640 root=0xcae..a8f3
INFO 21:08:41.069 [Fork choice] Adding new block slot=6000641 root=0xe4e..0178
INFO 21:08:41.070 [Block processing] Computing committees for epoch 187520 
INFO 21:08:41.264 [Fork choice] Block processed. Recomputing head. 
INFO 21:08:41.270 [Fork choice] Added new block slot=6000641 root=0xe4e..0178
INFO 21:08:41.270 [Fork choice] Adding new block slot=6000642 root=0x179..fdb5
INFO 21:08:41.430 [Fork choice] Block processed. Recomputing head. 
INFO 21:08:41.434 [Fork choice] Added new block slot=6000642 root=0x179..fdb5
INFO 21:08:41.434 [Fork choice] Adding new block slot=6000643 root=0xc1b..dc98
INFO 21:08:41.598 [Fork choice] Block processed. Recomputing head. 
INFO 21:08:41.604 [Fork choice] Added new block slot=6000643 root=0xc1b..dc98
INFO 21:08:48.004 [Libp2p] Slot transition slot=6000644
INFO 21:08:48.255 [Gossip] Block received, block.slot: 6000644. 
INFO 21:08:49.319 [Fork choice] Adding new block slot=6000644 root=0x4b3..ace3
INFO 21:08:49.521 [Fork choice] Block processed. Recomputing head. 
INFO 21:08:49.526 [Fork choice] Added new block slot=6000644 root=0x4b3..ace3
INFO 21:08:49.899 [Fork choice] Adding new block slot=6000644 root=0x4b3..ace3
INFO 21:08:50.097 [Fork choice] Block processed. Recomputing head. 
INFO 21:08:50.103 [Fork choice] Added new block slot=6000644 root=0x4b3..ace3
INFO 21:09:00.016 [Libp2p] Slot transition slot=6000645
INFO 21:09:00.111 [Gossip] Block received, block.slot: 6000645. 
INFO 21:09:00.160 [Fork choice] Adding new block slot=6000645 root=0xba5..6f18
INFO 21:09:00.348 [Fork choice] Block processed. Recomputing head. 
INFO 21:09:00.353 [Fork choice] Added new block slot=6000645 root=0xba5..6f18
INFO 21:09:01.156 [Fork choice] Adding new block slot=6000645 root=0xba5..6f18
INFO 21:09:01.332 [Fork choice] Block processed. Recomputing head. 
INFO 21:09:01.338 [Fork choice] Added new block slot=6000645 root=0xba5..6f18
INFO 21:09:12.003 [Libp2p] Slot transition slot=6000646
INFO 21:09:12.131 [Gossip] Block received, block.slot: 6000646. 
INFO 21:09:12.612 [Fork choice] Adding new block slot=6000646 root=0xb72..2380
INFO 21:09:12.801 [Fork choice] Block processed. Recomputing head. 
INFO 21:09:12.819 [Fork choice] Added new block slot=6000646 root=0xb72..2380
INFO 21:09:24.003 [Libp2p] Slot transition slot=6000647
INFO 21:09:24.374 [Gossip] Block received, block.slot: 6000647. 
INFO 21:09:24.390 [Fork choice] Adding new block slot=6000647 root=0x495..2721
INFO 21:09:24.577 [Fork choice] Block processed. Recomputing head. 
INFO 21:09:24.594 [Fork choice] Added new block slot=6000647 root=0x495..2721
INFO 21:09:25.237 [Fork choice] Adding new block slot=6000646 root=0xb72..2380
INFO 21:09:25.420 [Fork choice] Block processed. Recomputing head. 
INFO 21:09:25.423 [Fork choice] Added new block slot=6000646 root=0xb72..2380
WARNING 21:09:35.963 [Gossip] Block ignored, reason: "Block is from the future: block.slot=6000648. Current slot: 6000647.". 
INFO 21:09:36.002 [Libp2p] Slot transition slot=6000648
WARNING 21:09:47.966 [Gossip] Block ignored, reason: "Block is from the future: block.slot=6000649. Current slot: 6000648.". 
INFO 21:09:48.003 [Libp2p] Slot transition slot=6000649
INFO 21:10:00.011 [Libp2p] Slot transition slot=6000650
INFO 21:10:00.142 [Gossip] Block received, block.slot: 6000650. 
INFO 21:10:02.794 [Fork choice] Adding new block slot=6000648 root=0xcbd..5ec0
INFO 21:10:02.965 [Fork choice] Block processed. Recomputing head. 
INFO 21:10:02.971 [Fork choice] Added new block slot=6000648 root=0xcbd..5ec0
INFO 21:10:02.971 [Fork choice] Adding new block slot=6000649 root=0x353..3b52
INFO 21:10:03.129 [Fork choice] Block processed. Recomputing head. 
INFO 21:10:03.135 [Fork choice] Added new block slot=6000649 root=0x353..3b52
INFO 21:10:03.135 [Fork choice] Adding new block slot=6000650 root=0x7ec..d29f
INFO 21:10:03.293 [Fork choice] Block processed. Recomputing head. 
INFO 21:10:03.298 [Fork choice] Added new block slot=6000650 root=0x7ec..d29f
INFO 21:10:08.735 [Fork choice] Adding new block slot=6000650 root=0x7ec..d29f
INFO 21:10:08.924 [Fork choice] Block processed. Recomputing head. 
INFO 21:10:08.936 [Fork choice] Added new block slot=6000650 root=0x7ec..d29f
INFO 21:10:12.004 [Libp2p] Slot transition slot=6000651
INFO 21:10:12.209 [Gossip] Block received, block.slot: 6000651. 
INFO 21:10:12.308 [Fork choice] Adding new block slot=6000651 root=0x696..062b
INFO 21:10:12.496 [Fork choice] Block processed. Recomputing head. 
INFO 21:10:12.502 [Fork choice] Added new block slot=6000651 root=0x696..062b
INFO 21:10:19.133 [Fork choice] Adding new block slot=6000651 root=0x696..062b
INFO 21:10:19.324 [Fork choice] Block processed. Recomputing head. 
INFO 21:10:19.331 [Fork choice] Added new block slot=6000651 root=0x696..062b
INFO 21:10:24.005 [Libp2p] Slot transition slot=6000652
INFO 21:10:24.010 [Gossip] Block received, block.slot: 6000652. 
INFO 21:10:24.170 [Fork choice] Adding new block slot=6000652 root=0xd45..eb8d
INFO 21:10:24.360 [Fork choice] Block processed. Recomputing head. 
INFO 21:10:24.366 [Fork choice] Added new block slot=6000652 root=0xd45..eb8d
INFO 21:10:36.003 [Libp2p] Slot transition slot=6000653
INFO 21:10:36.116 [Gossip] Block received, block.slot: 6000653. 
INFO 21:10:36.122 [Fork choice] Adding new block slot=6000653 root=0x16d..b05a
INFO 21:10:36.293 [Fork choice] Block processed. Recomputing head. 
INFO 21:10:36.296 [Fork choice] Added new block slot=6000653 root=0x16d..b05a
INFO 21:10:37.443 [Fork choice] Adding new block slot=6000653 root=0x16d..b05a
INFO 21:10:37.635 [Fork choice] Block processed. Recomputing head. 
INFO 21:10:37.640 [Fork choice] Added new block slot=6000653 root=0x16d..b05a
INFO 21:10:41.623 [Fork choice] Adding new block slot=6000652 root=0xd45..eb8d
INFO 21:10:41.824 [Fork choice] Block processed. Recomputing head. 
INFO 21:10:41.830 [Fork choice] Added new block slot=6000652 root=0xd45..eb8d
INFO 21:10:48.005 [Libp2p] Slot transition slot=6000654
INFO 21:10:49.146 [Gossip] Block received, block.slot: 6000654. 
INFO 21:10:49.153 [Fork choice] Adding new block slot=6000654 root=0xa0d..f5b5
INFO 21:10:49.350 [Fork choice] Block processed. Recomputing head. 
INFO 21:10:49.357 [Fork choice] Added new block slot=6000654 root=0xa0d..f5b5
INFO 21:11:00.018 [Libp2p] Slot transition slot=6000655
INFO 21:11:00.097 [Gossip] Block received, block.slot: 6000655. 
INFO 21:11:01.462 [Fork choice] Adding new block slot=6000655 root=0x04a..81c5
INFO 21:11:01.661 [Fork choice] Block processed. Recomputing head. 
INFO 21:11:01.668 [Fork choice] Added new block slot=6000655 root=0x04a..81c5
WARNING 21:11:11.831 [Gossip] Block ignored, reason: "Block is from the future: block.slot=6000656. Current slot: 6000655.". 
INFO 21:11:12.004 [Libp2p] Slot transition slot=6000656
INFO 21:11:24.004 [Libp2p] Slot transition slot=6000657
INFO 21:11:24.389 [Gossip] Block received, block.slot: 6000657. 
INFO 21:11:24.841 [Fork choice] Adding new block slot=6000655 root=0x04a..81c5
INFO 21:11:25.036 [Fork choice] Block processed. Recomputing head. 
INFO 21:11:25.040 [Fork choice] Added new block slot=6000655 root=0x04a..81c5
WARNING 21:11:35.773 [Gossip] Block ignored, reason: "Block is from the future: block.slot=6000658. Current slot: 6000657.". 
INFO 21:11:36.003 [Libp2p] Slot transition slot=6000658
INFO 21:11:48.004 [Libp2p] Slot transition slot=6000659
INFO 21:11:50.920 [Fork choice] Adding new block slot=6000656 root=0xa2a..bd04
INFO 21:11:51.105 [Fork choice] Block processed. Recomputing head. 
INFO 21:11:51.117 [Fork choice] Added new block slot=6000656 root=0xa2a..bd04
INFO 21:11:51.117 [Fork choice] Adding new block slot=6000657 root=0x779..b77d
INFO 21:11:51.286 [Fork choice] Block processed. Recomputing head. 
INFO 21:11:51.288 [Fork choice] Added new block slot=6000657 root=0x779..b77d
INFO 21:12:00.015 [Libp2p] Slot transition slot=6000660
INFO 21:12:01.867 [Gossip] Block received, block.slot: 6000660. 
WARNING 21:12:11.999 [Gossip] Block ignored, reason: "Block is from the future: block.slot=6000661. Current slot: 6000660.". 
INFO 21:12:12.004 [Libp2p] Slot transition slot=6000661
INFO 21:12:17.403 [Fork choice] Adding new block slot=6000658 root=0x68d..ab72
INFO 21:12:17.599 [Fork choice] Block processed. Recomputing head. 
INFO 21:12:17.602 [Fork choice] Added new block slot=6000658 root=0x68d..ab72
INFO 21:12:17.602 [Fork choice] Adding new block slot=6000659 root=0x369..52ab
INFO 21:12:17.690 [Fork choice] Block processed. Recomputing head. 
INFO 21:12:17.694 [Fork choice] Added new block slot=6000659 root=0x369..52ab
INFO 21:12:17.694 [Fork choice] Adding new block slot=6000660 root=0x9e8..c9e0
INFO 21:12:17.858 [Fork choice] Block processed. Recomputing head. 
INFO 21:12:17.860 [Fork choice] Added new block slot=6000660 root=0x9e8..c9e0
INFO 21:12:24.005 [Libp2p] Slot transition slot=6000662
INFO 21:12:24.311 [Gossip] Block received, block.slot: 6000662. 
INFO 21:12:26.048 [Fork choice] Adding new block slot=6000661 root=0xeb9..7354
INFO 21:12:26.241 [Fork choice] Block processed. Recomputing head. 
INFO 21:12:26.252 [Fork choice] Added new block slot=6000661 root=0xeb9..7354
INFO 21:12:26.252 [Fork choice] Adding new block slot=6000662 root=0xf6e..e9c4
INFO 21:12:26.406 [Fork choice] Block processed. Recomputing head. 
INFO 21:12:26.412 [Fork choice] Added new block slot=6000662 root=0xf6e..e9c4
INFO 21:12:29.863 [Fork choice] Adding new block slot=6000662 root=0xf6e..e9c4
INFO 21:12:30.044 [Fork choice] Block processed. Recomputing head. 
INFO 21:12:30.052 [Fork choice] Added new block slot=6000662 root=0xf6e..e9c4
WARNING 21:12:35.826 [Gossip] Block ignored, reason: "Block is from the future: block.slot=6000663. Current slot: 6000662.". 
INFO 21:12:36.005 [Libp2p] Slot transition slot=6000663
INFO 21:12:48.005 [Libp2p] Slot transition slot=6000664
WARNING 21:12:59.967 [Gossip] Block ignored, reason: "Block is from the future: block.slot=6000665. Current slot: 6000664.". 
INFO 21:13:00.007 [Libp2p] Slot transition slot=6000665
INFO 21:13:12.003 [Libp2p] Slot transition slot=6000666
INFO 21:13:12.626 [Gossip] Block received, block.slot: 6000666. 
INFO 21:13:24.005 [Libp2p] Slot transition slot=6000667
INFO 21:13:25.009 [Gossip] Block received, block.slot: 6000667. 
INFO 21:13:25.421 [Fork choice] Adding new block slot=6000663 root=0x8ce..7ff7
INFO 21:13:25.617 [Fork choice] Block processed. Recomputing head. 
INFO 21:13:25.624 [Fork choice] Added new block slot=6000663 root=0x8ce..7ff7
INFO 21:13:25.625 [Fork choice] Adding new block slot=6000665 root=0xfcf..644e
INFO 21:13:25.739 [Fork choice] Block processed. Recomputing head. 
INFO 21:13:25.742 [Fork choice] Added new block slot=6000665 root=0xfcf..644e
INFO 21:13:25.743 [Fork choice] Adding new block slot=6000666 root=0xdd8..6738
INFO 21:13:25.912 [Fork choice] Block processed. Recomputing head. 
INFO 21:13:25.926 [Fork choice] Added new block slot=6000666 root=0xdd8..6738
INFO 21:13:25.927 [Fork choice] Adding new block slot=6000667 root=0xde3..990c
INFO 21:13:26.080 [Fork choice] Block processed. Recomputing head. 
INFO 21:13:26.086 [Fork choice] Added new block slot=6000667 root=0xde3..990c
INFO 21:13:31.392 [Fork choice] Adding new block slot=6000667 root=0xde3..990c
INFO 21:13:31.590 [Fork choice] Block processed. Recomputing head. 
INFO 21:13:31.601 [Fork choice] Added new block slot=6000667 root=0xde3..990c
WARNING 21:13:35.832 [Gossip] Block ignored, reason: "Block is from the future: block.slot=6000668. Current slot: 6000667.". 
INFO 21:13:36.003 [Libp2p] Slot transition slot=6000668
INFO 21:13:48.005 [Libp2p] Slot transition slot=6000669
INFO 21:13:48.774 [Gossip] Block received, block.slot: 6000669. 
INFO 21:14:00.015 [Libp2p] Slot transition slot=6000670
INFO 21:14:00.061 [Gossip] Block received, block.slot: 6000670. 
INFO 21:14:02.431 [Fork choice] Adding new block slot=6000668 root=0xa51..14df
INFO 21:14:02.630 [Fork choice] Block processed. Recomputing head. 
INFO 21:14:02.636 [Fork choice] Added new block slot=6000668 root=0xa51..14df
INFO 21:14:02.637 [Fork choice] Adding new block slot=6000669 root=0xfda..8fef
INFO 21:14:02.726 [Fork choice] Block processed. Recomputing head. 
INFO 21:14:02.733 [Fork choice] Added new block slot=6000669 root=0xfda..8fef
INFO 21:14:02.733 [Fork choice] Adding new block slot=6000670 root=0xe60..aa6b
INFO 21:14:02.891 [Fork choice] Block processed. Recomputing head. 
INFO 21:14:02.903 [Fork choice] Added new block slot=6000670 root=0xe60..aa6b
INFO 21:14:09.209 [Fork choice] Adding new block slot=6000670 root=0xe60..aa6b
INFO 21:14:09.396 [Fork choice] Block processed. Recomputing head. 
INFO 21:14:09.403 [Fork choice] Added new block slot=6000670 root=0xe60..aa6b
INFO 21:14:12.008 [Libp2p] Slot transition slot=6000671
INFO 21:14:12.585 [Gossip] Block received, block.slot: 6000671. 
INFO 21:14:12.596 [Fork choice] Adding new block slot=6000671 root=0x8c6..d57e
INFO 21:14:12.791 [Fork choice] Block processed. Recomputing head. 
INFO 21:14:12.797 [Fork choice] Added new block slot=6000671 root=0x8c6..d57e
INFO 21:14:24.004 [Libp2p] Slot transition slot=6000672
INFO 21:14:27.545 [Gossip] Block received, block.slot: 6000672. 
INFO 21:14:27.718 [Fork choice] Adding new block slot=6000672 root=0x611..070f
INFO 21:14:28.073 [Fork choice] Block processed. Recomputing head. 
ERROR 21:14:28.076 GenServer LambdaEthereumConsensus.Libp2pPort terminating
** (RuntimeError) Parent CAE1F13DBD7FAF2177FDEE036E00B1D59319D380A9DC859478940CB1B0FCA8F3 not found in tree
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/simple_tree.ex:84: LambdaEthereumConsensus.ForkChoice.Simple.Tree.get_children!/2
    (lambda_ethereum_consensus 0.1.0) lib/types/store.ex:148: Types.Store.get_children/2
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/head.ex:90: LambdaEthereumConsensus.ForkChoice.Head.filter_block_tree/4
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/head.ex:85: LambdaEthereumConsensus.ForkChoice.Head.get_filtered_block_tree/1
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/head.ex:14: LambdaEthereumConsensus.ForkChoice.Head.get_head/1
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:272: LambdaEthereumConsensus.ForkChoice.recompute_head/1
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/fork_choice.ex:62: anonymous fn/1 in LambdaEthereumConsensus.ForkChoice.on_block/2
    (telemetry 1.3.0) /Users/roliveri/prog/elixir/consensus/lambda_ethereum_consensus/deps/telemetry/src/telemetry.erl:324: :telemetry.span/3
Last message: {#Port<0.19>, {:data, <<10, 152, 76, 10, 40, 47, 101, 116, 104, 50, 47, 100, 51, 49, 102, 54, 49, 57, 49, 47, 98, 108, 111, 98, 95, 115, 105, 100, 101, 99, 97, 114, 95, 49, 47, 115, 115, 122, 95, 115, 110, 97, 112, 112, 121, 26, ...>>}} 
INFO 21:14:28.106 [Optimistic Sync] Waiting 10.0 seconds to discover some peers before requesting blocks. 
INFO 21:14:28.107 [Fork choice] Adding new block slot=6000672 root=0x611..070f
```

**Comment by rodrigo-o** [comment]
In #1316 I was able to check that the store is in al old state and that cause the setting of the block as invalid:

```
INFO 16:00:48.604 [Fork choice] Adding new block slot=6013504 root=0xaa1..c455
INFO 16:00:48.802 [Fork choice] Block processed. Recomputing head. 
ERROR 16:00:48.805 GenServer LambdaEthereumConsensus.Libp2pPort terminating
** (RuntimeError) Parent 8C7A76877C9CD8052DFF5995378515B9140DF11467672AEFFA7D0E3A1B9AE05B not found in tree
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/simple_tree.ex:84: LambdaEthereumConsensus.ForkChoice.Simple.Tree.get_children!/2
    (lambda_ethereum_consensus 0.1.0) lib/types/store.ex:144: Types.Store.get_children/2
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/head.ex:90: LambdaEthereumConsensus.ForkChoice.Head.filter_block_tree/4
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/head.ex:85: LambdaEthereumConsensus.ForkChoice.Head.get_filtered_block_tree/1
    (lambda_ethereum_consensus 0.1.0) lib/lambda_ethereum_consensus/fork_choice/head.ex:14: LambdaEthereumConsensus.ForkChoice.Head.get_head/1
    (lambda_ethereum_consensus 0.1.0) iex:296: LambdaEthereumConsensus.ForkChoice.recompute_head/1
    (lambda_ethereum_consensus 0.1.0) iex:62: anonymous fn/1 in LambdaEthereumConsensus.ForkChoice.on_block/2
    (telemetry 1.3.0) /Users/roliveri/prog/elixir/consensus/lambda_ethereum_consensus/deps/telemetry/src/telemetry.erl:324: :telemetry.span/3
Last message: {#Port<0.19>, {:data, <<10, 174, 186, 1, 10, 38, 47, 101, 116, 104, 50, 47, 100, 51, 49, 102, 54, 49, 57, 49, 47, 98, 101, 97, 99, 111, 110, 95, 98, 108, 111, 99, 107, 47, 115, 115, 122, 95, 115, 110, 97, 112, 112, 121, 26, 20, ...>>}} 
INFO 16:00:48.828 [Optimistic Sync] Waiting 10.0 seconds to discover some peers before requesting blocks. 
INFO 16:00:48.828 [Fork choice] Adding new block slot=6013504 root=0xaa1..c455
INFO 16:00:48.881 Store slot: 6013457 is in the past, slot: 6013504 is in the future. Current chain slot: 6013504 
ERROR 16:00:48.881 [Fork choice] Failed to add block: block is from the future slot=6013504 root=0xaa1..c455
ERROR 16:00:48.882 [PendingBlocks] Saving block as invalid block is from the future slot=6013504 root=0xaa1..c455
```

Either way the root cause is the same, the tree manage in the Recomputing head

**Comment by rodrigo-o** [comment]
Resolved by #1318
